cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1861,9 +1879,12 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
     }
 
     // Handle null filtering.
-    if (rangeSize == 0 && !nullAllowed) {
-      std::unique_ptr<common::IsNotNull> filter = std::make_unique<common::IsNotNull>();
-      filters[common::Subfield(inputName)] = std::move(filter);
+    if (rangeSize == 0) {
+      if (!nullAllowed) {
+        filters[common::Subfield(inputName)] = std::move(std::make_unique<common::IsNotNull>());
+      } else if (isNull) {
+        filters[common::Subfield(inputName)] = std::move(std::make_unique<common::IsNull>());
+      }
```

Comment:
Do we also need to add this check here? https://github.com/oap-project/gluten/pull/3791/files#diff-08ec1ecb10e48cd471188daf2152228da4b00a81c16acaf2e87b040fec16f8c5R1843

Reply:
Updated, thanks.